### PR TITLE
Editor redo undo fix

### LINKED ISSUE #834: Editor undo redo steps get messed up
In some situations the undo redo steps get messed up.
Probably caused by doing an action after undoing something, the root issue here is maybe that the actions that should be discarded in that case aren't actually discarded.

### PR DESCRIPTION
**Pull request by Dak2896**
* Redo now check if we are making new action in the history or if it's just normal redo
* If action is new all redo history is erased from that point

fixes #834

### CODE REVIEW

**Comment by hhyyrylainen** [comment]
Tests are failing due to warning from AS about (Signed/Unsigned mismatch):
```
[SCRIPT] [WARNING] /root/thrive/build/bin/Data/Scripts/microbe_stage/microbe_editor/microbe_editor.as (624, 56) : Signed/Unsigned mismatch
```

File: scripts/microbe_stage/microbe_editor/microbe_editor.as [hunk 1]
```diff
@@ -617,18 +617,36 @@ class MicrobeEditor{
...
+
+            int q = int(action.data["q"]);
+            int r = int(action.data["r"]);
+
+            int nextQ = int(actionHistory[actionIndex].data["q"]);
+            int nextR = int(actionHistory[actionIndex].data["r"]);
+
+            int i =  actionIndex;
+
+            if( (q != nextQ) || (r != nextR)) {
```
Comment (hhyyrylainen) [inline scripts/microbe_stage/microbe_editor/microbe_editor.as]: I'm failing to undestand why this is needed? Wouldn't logic like this work: if actionIndex is not at the end of action History, chop off items from actionHistory until it is at the end?

Comment (Dak2896) [reply scripts/microbe_stage/microbe_editor/microbe_editor.as]: You right, that check is useless, i commit the changes soon.

File: scripts/microbe_stage/microbe_editor/microbe_editor.as [hunk 2]
```diff
@@ -617,18 +617,22 @@ class MicrobeEditor{
...
                 return;
-        // We resize always and insert at the index so that
-        // if we undo something then add something, the new
-        // action is in the right spot on the array.
-        // since we only enqueue when we add new actions
-        actionHistory.resize(actionHistory.length()+1);
+
+        // Every time we make a new action if is in the middle of history of actions
+        // We have to erase all redo history from that point
+        if( actionIndex < actionHistory.length() ) {
```
Comment (hhyyrylainen) [inline scripts/microbe_stage/microbe_editor/microbe_editor.as]: Using a while loop calling actionHistory.removeLast could remove this and be cleaner:
```
while(actionIndex < actionHistory.length())
    actionHistory.removeLast();
```

**Comment by Dak2896** [comment]
Still same error, like `actionIndex` is not same type of `actionHistory.length()`

**Comment by hhyyrylainen** [comment]
They are not the same type. I'll commit a fix.

**Comment by Dak2896** [comment]
Oh, ok, just a simple cast fix it.

**Comment by hhyyrylainen** [comment]
I think the undo/redo is fixed with this change. I'll merge once the tests pass. Unless you have something you want to add?

**Comment by Dak2896** [comment]
No there is nothing more to add on this branch

**Pull request merged**